pkg/secretlock/local/standard_secret_lock.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package local
+
+import (
+	"bytes"
+	"crypto/cipher"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/google/tink/go/subtle/random"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+)
+
+// package local provides a local secret lock service. The user must create a master key and store it
+// in a local file or an environment variable prior to using this service.
+//
+// A utility function is offered to help create an encrypted master key using a passphrase called:
+// 		CreateMasterKey(passphrase)
+//
+// The returned encrypted master key must be stored either in a file or an environment variable.
+// The user can then call either:
+//		InitFromPath(passphrase, path) or
+//		InitFromEnv(passphrase, envPrefix, keyURI)
+// to get an authenticated io.Reader instance needed to create a Lock service.
+//
+// To get the lock service, call:
+// 		NewServiceFromReader(passphrase, reader)
+// where reader is the instance returned from one of the previous Init functions.
+
+var logger = log.New("aries-framework/lock")
+
+// Lock is a secret lock service responsible for encrypting keys using a master key stored as environment variable.
+type Lock struct {
+	aead cipher.AEAD
+}
+
+// InitFromPath creates a new instance of a local secret lock Reader for an encrypted master key in path.
+func InitFromPath(passphrase, path string) (io.Reader, error) {
+	masterKeyFile, err := os.OpenFile(path, os.O_RDONLY, 0600)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		err = masterKeyFile.Close()
+		if err != nil {
+			logger.Warnf("failed to close file: %w", err)
+		}
+	}()
+
+	mkData := make([]byte, 512)
+
+	n, err := io.ReadFull(masterKeyFile, mkData)
+	if err != io.ErrUnexpectedEOF {
+		return nil, err
+	}
+
+	mkData = mkData[0:n]
+
+	return newReader(passphrase, base58.Decode(string(mkData)))
+}
+
+// InitFromEnv creates a new instance of a local secret lock Reader for an encrypted master key found in env variable.
+func InitFromEnv(passphrase, envPrefix, keyURI string) (io.Reader, error) {
+	mk := os.Getenv(envPrefix + strings.ReplaceAll(keyURI, "/", "_"))
+	if mk == "" {
+		return nil, fmt.Errorf("masterKey not set")
+	}
+
+	return newReader(passphrase, base58.Decode(mk))
+}
+
+// NewServiceFromReader creates a new instance of local secret lock service for a master key reader.
```

Comment:
s/for a master key reader/from a master key reader/

Reply:
I'll s/secret lock service for a master key reader/secret lock service using a master key in reader/

Reply:
this function was refactored